start.go

```diff
@@ -121,18 +132,63 @@ func startContainer(context *cli.Context, spec *specs.LinuxSpec, rspec *specs.Li
 			process.ExtraFiles = append(process.ExtraFiles, os.NewFile(uintptr(i), ""))
 		}
 	}
-	tty, err := newTty(spec.Process.Terminal, process, rootuid, context.String("console"))
-	if err != nil {
-		return -1, err
+	var tty *tty
+	if spec.Process.Terminal {
+		if tty, err = createTty(process, rootuid, context.String("console")); err != nil {
+			return -1, err
+		}
+	} else if detach {
+		if err := dupStdio(process, rootuid); err != nil {
+			return -1, err
+		}
+	} else {
+		if tty, err = createStdioPipes(process, rootuid); err != nil {
+			return -1, err
+		}
 	}
-	handler := newSignalHandler(tty)
-	defer handler.Close()
 	if err := container.Start(process); err != nil {
 		return -1, err
 	}
+	if pidFile := context.String("pid-file"); pidFile != "" {
+		pid, err := process.Pid()
+		if err != nil {
+			return -1, err
+		}
+		f, err := os.Create(pidFile)
+		if err != nil {
+			logrus.WithField("pid", pid).Error("create pid file")
+		} else {
+			_, err = fmt.Fprintf(f, "%d", pid)
+			f.Close()
+			if err != nil {
+				logrus.WithField("error", err).Error("write pid file")
+			}
+		}
+	}
+	if detach {
+		return 0, nil
+	}
+	handler := newSignalHandler(tty)
+	defer handler.Close()
 	return handler.forward(process)
 }
 
+func dupStdio(process *libcontainer.Process, rootuid int) error {
+	process.Stdin = os.Stdin
```

Comment:
So is the difference between this and ```createStdioPipes``` that the host's FD's are passed in vs new pipes created?